Library/Homebrew/dev-cmd/tap-new.rb

```diff
@@ -53,23 +53,36 @@ def tap_new
         pull_request: []
       jobs:
         test-bot:
-          runs-on: macos-latest
+          runs-on: [ubuntu-latest, macos-latest]
           steps:
+            - name: Update Homebrew
+              run: brew update
+
             - name: Set up Git repository
               uses: actions/checkout@v2
-            - name: Run brew test-bot
+
+            - name: Set up Homebrew
               run: |
-                set -e
-                brew update
                 HOMEBREW_TAP_DIR="/usr/local/Homebrew/Library/Taps/#{tap.full_name}"
                 mkdir -p "$HOMEBREW_TAP_DIR"
                 rm -rf "$HOMEBREW_TAP_DIR"
                 ln -s "$PWD" "$HOMEBREW_TAP_DIR"
-                brew test-bot
+
+            - name: Run brew test-bot --only-cleanup-before
+              run: brew test-bot --only-cleanup-before
+
+            - name: Run brew test-bot --only-setup
+              run: brew test-bot --only-setup
+
+            - name: Run brew test-bot --only-tap-syntax
+              run: brew test-bot --only-tap-syntax
+
+            - name: Run brew test-bot --only-formulae
+              run: brew test-bot --only-formulae
```

Comment:
`if: github.event_name == 'pull_request'` or similar so it doesn't happen on master pushes.